sdk/storage/azure-storage-blob/tests/test_container.py

```diff
@@ -1607,6 +1607,32 @@ def test_delete_blobs_simple(self, **kwargs):
         assert response[1].status_code == 202
         assert response[2].status_code == 202
 
+    @BlobPreparer()
+    @recorded_by_proxy
+    def test_delete_blob_with_properties_versioning(self, **kwargs):
+        versioned_storage_account_name = kwargs.pop("versioned_storage_account_name")
+        versioned_storage_account_key = kwargs.pop("versioned_storage_account_key")
+
+        bsc = BlobServiceClient(self.account_url(versioned_storage_account_name, "blob"), versioned_storage_account_key)
+        container: ContainerClient = self._create_container(bsc)
+
+        blob_name = self.get_resource_name("utcontainer")
+        blob_data = 'abc'
+        blob_client = container.get_blob_client(blob_name)
+
+        blob_client.upload_blob(blob_data, overwrite=True)
+        v1_props = blob_client.get_blob_properties()
+        blob_client.upload_blob(blob_data * 2, overwrite=True)
+        v2_props = blob_client.get_blob_properties()
+
+        # Act
+        container.delete_blob(v2_props, version_id=v1_props['version_id'])
```

Comment:
May be good to also include the other case where we extract version_id from the props. Here you are only testing the keyword overrides the properties but we should test the properties case too since that's what's being added.

I think you can do it in the same test but just doing another delete of `v2` using just `v2_props`?

Same with `test_download_blob_with_properties_versioning`

Reply:
Doi. Good catch. I generated another version since the current behavior was operating on the latest version, so I made sure it parsed out from properties on a non-latest version 😄